ReadStore::from requires that the underlying Lens is Writeable

**Problem**

When you have a ReadStore and use a lensing function to access one of its fields, that resulting store cannot be converted into another ReadStore.

Aka. you cannot map a ReadStore to a ReadStore

**Steps To Reproduce**

```rust
        #[derive(Store, Default)]
        struct Thing {
            entry: String,
        }
        let a = Store::new(Thing::default());
        let b = ReadStore::from(a);

        let b_entry = b.entry();
        let b_entry_read = ReadStore::from(b.entry());
```

```
22:41:52 [cargo] error[E0277]: the trait bound `dioxus::prelude::ReadSignal<Thing>: Writable` is not satisfied
   --> crates/story-teller-app/src/components/townsquare/seat_circle.rs:558:28
    |
558 |         let b_entry_read = ReadStore::from(b.entry());
    |                            ^^^^^^^^^ the trait `Writable` is not implemented for `dioxus::prelude::ReadSignal<Thing>`
```


**Expected behavior**

It seems to me that this should be possibe, as it feels like a common use case to receive a ReadStore in a component, and then pass a lensed ReadStore into a deeper component.


**Environment:**

- Dioxus version: v0.7.1
- Rust version:   0.97.0-nightly
- App platform:    web

**Questionnaire**

I'm interested in fixing this myself but don't know where to start.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -11,7 +11,7 @@ on:
       - main
 
   pull_request:
-    types: [opened, synchronize, reopened, ready_for_review]
+    types: [opened, synchronize, reopened, edited, ready_for_review]
     branches:
       - main
 

--- a/packages/stores/src/store.rs
+++ b/packages/stores/src/store.rs
@@ -204,7 +204,7 @@ where
 impl<__F, __FMut, T: ?Sized, S, Lens> ::std::convert::From<MappedStore<T, Lens, __F, __FMut>>
     for ReadStore<T, S>
 where
-    Lens: Writable<Storage = S> + 'static,
+    Lens: Readable<Storage = S> + 'static,
     __F: Fn(&Lens::Target) -> &T + 'static,
     __FMut: Fn(&mut Lens::Target) -> &mut T + 'static,
     S: BoxedSignalStorage<T> + CreateBoxedSignalStorage<MappedMutSignal<T, Lens, __F, __FMut>>,

